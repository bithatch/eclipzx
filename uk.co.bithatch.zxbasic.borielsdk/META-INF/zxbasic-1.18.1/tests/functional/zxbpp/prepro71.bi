#include "alloc.bas"

PRINT "HOLA"
