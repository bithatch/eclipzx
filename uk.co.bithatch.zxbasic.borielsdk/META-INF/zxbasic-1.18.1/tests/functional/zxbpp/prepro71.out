#line 1 "prepro71.bi"
#line 1 "/zxbasic/src/lib/arch/zx48k/stdlib/alloc.bas"











#line 13 "/zxbasic/src/lib/arch/zx48k/stdlib/alloc.bas"

#pragma push(case_insensitive)
#pragma case_insensitive = True















function FASTCALL allocate(byval n as uinteger) as uinteger
	
	
	
	
	asm
    push namespace core
    ld b, h
    ld c, l
    jp __MEM_ALLOC 
    pop namespace
	end asm
end function

















function FASTCALL callocate(byval n as uinteger) as uinteger
	
	
	
	
	asm
    push namespace core
    ld b, h
    ld c, l
    jp __MEM_CALLOC 
    pop namespace
	end asm
end function








sub FASTCALL deallocate(byval addr as integer)
	
	
	
	asm
    push namespace core
    jp __MEM_FREE
    pop namespace
	end asm
end sub
















function FASTCALL reallocate(byval addr as uinteger, byval n as uinteger) as uinteger
	
	
	
	
	
	asm
    push namespace core
    ex de, hl 
    pop hl    
    ex (sp), hl 
    ld b, h
    ld c, l
    ex de, hl 
    jp __REALLOC 
    pop namespace
	end asm
end function








function FASTCALL memavail as uInteger
    asm
    push namespace core
    PROC

    LOCAL LOOP

    ld hl, ZXBASIC_MEM_HEAP
    ld de, 0 

LOOP:
    
    ld c, (hl)
    inc hl
    ld b, (hl)
    inc hl

    
    ld a, (hl)
    inc hl
    ld h, (hl)
    ld l, a

    
    ex de, hl
    add hl, bc
    ex de, hl

    
    ld a, h
    or l
    jr nz, LOOP

    ex de, hl

    ENDP
    pop namespace
    end asm
end function







function FASTCALL maxavail as uInteger
    asm
    push namespace core
    PROC

    LOCAL LOOP, CONT

    ld hl, ZXBASIC_MEM_HEAP
    ld de, 0 

LOOP:
    
    ld c, (hl)
    inc hl
    ld b, (hl)
    inc hl

    
    ld a, (hl)
    inc hl
    ld h, (hl)
    ld l, a

    
    
    ex de, hl
    or a
    sbc hl, bc  
    add hl, bc  
    ex de, hl

    
    jr nc, CONT
    
    ld d, b
    ld e, c

CONT:
    
    ld a, h
    or l
    jr nz, LOOP

    ex de, hl

    ENDP
    pop namespace
    end asm
end function


#pragma pop(case_insensitive)

#require "alloc.asm"
#require "free.asm"
#require "realloc.asm"
#require "calloc.asm"

#line 239 "/zxbasic/src/lib/arch/zx48k/stdlib/alloc.bas"
#line 2 "prepro71.bi"

PRINT "HOLA"
